tests/python/mkl/test_mkldnn.py

```diff
@@ -383,6 +383,17 @@ def check_fullyconnected_training(stype):
     for stype in stypes:
         check_fullyconnected_training(stype)
 
+def test_softmax_with_large_negative_inputs():
+    input_data = mx.nd.array([[[[-1e30,-1e30]]]])
```

Comment:
Could you build other corner cases to cover more, such as 1e-30 as input or other possible combinations?

Reply:
Added more corner cases: extremely large positive and negative inputs covered (also checked for max and min values of float32 inputs). Please suggest if you think there are more corner cases that should be added.